src/components/graph/Graph.jsx

```diff
@@ -293,7 +298,16 @@ export default class Graph extends React.Component {
                 () => this.props.onClickNode && this.props.onClickNode(clickedNodeId)
             );
         } else {
-            this.props.onClickNode && this.props.onClickNode(clickedNodeId);
+            if (!this.timeoutID) {
+                this.timeoutID = setTimeout(() => {
```

Comment:
I would prefer the naming `this.nodeClickTimer`. Also, do you need to assign the timer to `this`?

Reply:
Yes, from what I've tried we can't initiate a variable (here, a nodeClickTimer not linked to this) in a component, or I would need to add a state but I found that too heavy for just one functionality